### PR TITLE
Wi-Fi Plugin: various fixes/improvements

### PR DESCRIPTION
**Pull request by sirDonovan**
Fix giveaways becoming frozen/endless from hotpatching, fix possible
crash in lottery giveaways, simplify lottery drawings, and improve
wording in multiple statements.

### CODE REVIEW

File: chat-plugins/wifi.js [hunk 1]
```diff
@@ -77,37 +74,30 @@ var QuestionGiveAway = (function () {
 
 		this.answered[userid]++;
 		if (this.answered[userid] >= 3) {
-			output.sendReply("Your guess " + guess + " is wrong. You have used up all your guesses. Better luck next time!");
+			output.sendReply("Your guess '" + guess + "' is wrong. You have used up all of your guesses, better luck next time!");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: Nice comma splice.

File: chat-plugins/wifi.js [hunk 2]
```diff
@@ -77,37 +74,30 @@ var QuestionGiveAway = (function () {
...
 		} else {
-			output.sendReply("Your guess " + guess + " is wrong. Try again!");
+			output.sendReply("Your guess '" + guess + "' is wrong, try again!");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: Here too.

File: chat-plugins/wifi.js [hunk 3]
```diff
@@ -77,37 +74,30 @@ var QuestionGiveAway = (function () {
...
 			this.question = value;
 			output.sendReply("The question has been changed to " + value + ".");
-			break;
-		case 'answer':
+		} else {
 			var ans = this.sanitizeAnswers(value);
-			if (Object.keys(ans).length <= 0) return output.sendReply("You must specify at least one answer and must not contain any special characters.");
-
+			var len = Object.keys(ans).length;
+			if (!len) return output.sendReply("You must specify at least one answer and must not contain any special characters.");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: "and it must not" imo

File: chat-plugins/wifi.js [hunk 4]
```diff
@@ -116,27 +106,24 @@ var QuestionGiveAway = (function () {
...
+			if (!this.winner) {
+				this.room.addRaw("<b>The giveaway has been forcibly ended, as no one has answered the question.</b>");
+			} else {
+				var ans = [];
+				for (var i in this.answers) {
+					ans.push(this.answers[i]);
+				}
+				this.room.addRaw("<div class='broadcast-blue'><b>" + Tools.escapeHTML(this.winner.name) + "</b> guessed the correct answer.</b> Congratulations!<br/>" +
+					"Correct answer(s): " + ans.join(','));
+				if (this.winner.connected) this.winner.popup('You have won the giveaway. PM **' + Tools.escapeHTML(this.giver.name) + '** to claim your reward!');
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: change reward to prize

File: chat-plugins/wifi.js [hunk 5]
```diff
@@ -228,72 +208,59 @@ var LotteryGiveAway = (function () {
 		if (force) {
 			if (this.phase === 'ended') return;
 			clearTimeout(this.drawTimer);
-			this.room.addRaw('<b>The giveaway was forcibly ended, as enough users did not participate.</b>');
+			this.room.addRaw("<b>The giveaway was forcibly ended, as not enough users participated.</b>");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: remove the comma

File: chat-plugins/wifi.js [hunk 6]
```diff
@@ -301,17 +268,19 @@ var commands = {
...
 		if (room.id !== 'wifi') return false;
-		if (!giveaway) return this.sendReply('There is no giveaway going on at the moment.');
-		if (giveaway.type !== 'question') return this.sendReply('This is not a question giveaway.');
-		if (!target.trim()) return this.sendReply("You must mention a question/answer.");
+		if (!giveaways[room.id]) return this.sendReply("There is no giveaway going on at the moment.");
+		if (giveaways[room.id].type !== 'question') return this.sendReply("This is not a question giveaway.");
 
-		giveaway.change(cmd.substr(6), target.trim(), user, this);
+		target = target.trim();
+		if (!target) return this.sendReply("You must include a question/answer.");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: "question and a answer"

Comment (sirDonovan) [inline chat-plugins/wifi.js]: Went with 'or' instead of 'and', as this command can't change both at once.

File: chat-plugins/wifi.js [hunk 7]
```diff
@@ -116,27 +106,24 @@ var QuestionGiveAway = (function () {
...
-			this.room.addRaw('<b>The giveaway has been forcibly ended, as no one has answered the question.	</b>');
+			this.room.addRaw("<b>The giveaway was forcibly ended.</b>");
 		} else {
-			var ans = [];
-			for (var i in this.answers) {
-				ans.push(this.answers[i]);
+			this.phase = 'ended';
+			clearTimeout(this.endTimer);
+			if (!this.winner) {
+				this.room.addRaw("<b>The giveaway has been forcibly ended, as no one has answered the question.</b>");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: Remove comma

File: chat-plugins/wifi.js [hunk 8]
```diff
@@ -77,37 +74,30 @@ var QuestionGiveAway = (function () {
...
 	};
 
 	QuestionGiveAway.prototype.change = function (key, value, user, output) {
-		if (user.userid !== this.host.userid) return this.sendReply("You cannot edit this giveaway. Only the host/starter can edit.");
-		if (this.phase !== 'pending') return output.sendReply("The giveaway has started, and the question/answer cannot be changed.");
-
-		switch (key) {
-		case 'question':
+		if (user.userid !== this.host.userid) return this.sendReply("Only the host can edit the giveaway.");
+		if (this.phase !== 'pending') return output.sendReply("You cannot change the question/answer once the giveaway has started.");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: "or" instead of /

File: chat-plugins/wifi.js [hunk 9]
```diff
@@ -161,14 +148,14 @@ var LotteryGiveAway = (function () {
...
 
-		this.reminder = '<center><div class = "broadcast-blue"><font size = 3><b>It\'s giveaway time!</b></font><br/>' +
-			'<font size = 1>Giveaway started by ' + Tools.escapeHTML(host.name) + '</font><br/><br/>' +
+		this.reminder = '<center><div class="broadcast-blue"><font size="3"><b>It\'s giveaway time!</b></font><br/>' +
+			'<font size="1">Giveaway started by ' + Tools.escapeHTML(host.name) + '</font><br/><br/>' +
 			'<b>' + Tools.escapeHTML(giver.name) + '</b> will be giving away: <b>' + Tools.escapeHTML(this.prize) + '</b>!<br/>' +
-			'The lottery drawing will occur in 2 minutes, with ' + this.maxwinners + ' winners!<br/>' +
-			'<button name = "send" value = "/giveaway joinlottery"><font size = 1><b>Join</b></font></button> <button name = "send" value = "/giveaway leavelottery"><font size = 1><b>Leave</b></font></button><br/>' +
-			'<font size = 1><b><u>Note:</u> Please do not join if you don\'t have a 3DS and a copy of Pokémon XY or ORAS';
+			'The lottery drawing will occur in 2 minutes, with ' + this.maxwinners + ' winner' + (this.maxwinners > 1 ? 's' : '') + '!<br/>' +
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: "and with". can keep the comma.

File: chat-plugins/wifi.js [hunk 10]
```diff
@@ -228,72 +208,59 @@ var LotteryGiveAway = (function () {
...
 		target = target.split(',').map(function (val) { return val.trim(); });
-		if (!target[0] || !target[1] || !target[2]) return this.sendReply('Too few parameters.');
+		if (target.length !== 3) return this.sendReply("Invalid arguments specified - /question giver, prize, question, answer(s)");
 		var options = {
 			prize: target[0],
 			question: target[1],
 			answers: QuestionGiveAway.sanitizeAnswers(target[2])
 		};
-		if (Object.keys(options.answers).length <= 0) return this.sendReply("You must specify at least one answer and cannot contain any special characters.");
+		if (!Object.keys(options.answers).length) return this.sendReply("You must specify at least one answer and cannot contain any special characters.");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: and it

File: chat-plugins/wifi.js [hunk 11]
```diff
@@ -325,32 +294,29 @@ var commands = {
...
 
 		target = target.split(',').map(function (val) { return val.trim(); });
-		if (!target[0] || !target[1]) return this.sendReply('Too few parameters.');
+		if (target.length !== 2) return this.sendReply("Invalid arguments specified - /lottery giver, prize, max winners");
 		var options = {
 			prize: target[0],
 			maxwinners: parseInt(target[1])
 		};
-		if (options.maxwinners > 10 || options.maxwinners < 1) return this.sendReply("The lottery giveaway can have a minimum of 1 and maximum of 10 winners.");
+		if (options.maxwinners > 10 || options.maxwinners < 1 || isNaN(options.maxwinners)) return this.sendReply("The lottery giveaway can have a minimum of 1 and maximum of 10 winners.");
```
Comment (TheImmortal) [inline chat-plugins/wifi.js]: and a

**Pull request merged**